rust/ql/lib/codeql/rust/security/SqlInjectionExtensions.qll

```diff
@@ -0,0 +1,50 @@
+/**
+ * Provides classes and predicates for reasoning about database
+ * queries built from user-controlled sources (that is, SQL injection
+ * vulnerabilities).
+ */
+
+import rust
+private import codeql.rust.dataflow.DataFlow
+private import codeql.rust.Concepts
+private import codeql.util.Unit
+
+/**
+ * Provides default sources, sinks and barriers for detecting SQL injection
+ * vulnerabilities, as well as extension points for adding your own.
+ */
+module SqlInjection {
+  /**
+   * A data flow source for SQL injection vulnerabilities.
+   */
+  abstract class Source extends DataFlow::Node { }
+
+  /**
+   * A data flow sink for SQL injection vulnerabilities.
+   */
+  abstract class Sink extends DataFlow::Node { }
+
+  /**
+   * A barrier for SQL injection vulnerabilities.
+   */
+  abstract class Barrier extends DataFlow::Node { }
+
+  /**
+   * An active threat-model source, considered as a flow source.
+   */
+  private class ActiveThreatModelSourceAsSource extends Source, ActiveThreatModelSource { }
```

Comment:
Shouldn't this be restricted somehow using `getSourceType()`?

Reply:
I don't think so.  `ActiveThreatModelSource` is a restriction of `ThreatModelSource` according to user preferences (however that works), which defaults to just the remote sources I think.  This sounds like exactly what we want.

Reply:
Ah ok, to `ActiveThreatModelSource` should be used for all queries that already use remote flow sources?

Reply:
I believe that's the idea.  Unless you have a special reason for *really* only wanting remote flow sources, but generally in the past we've generally specified remote only because most users aren't interested in flow from argv